test/distributed/test_functional_api.py

```diff
@@ -864,6 +864,66 @@ def run_with_backward():
 
         self.assertIsNotNone(t.grad)
 
+    @parametrize("compile", [True, False])
+    def test_all_gather_tensor(self, compile: bool) -> None:
```

Comment:
nit: consider renaming this to sth like `test_all_gather_tensor_with_grad` as I believe there were already some similar tests that test the API without grad?

Reply:
The test is `TestFunctionalAutograd::test_all_gather_tensor` -- the grad is in the class name. Do you think it's worth adding it to each test as well?